### PR TITLE
[SPARK-40888][SQL][TESTS] Check error classes in HiveQuerySuite

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This PR aims to replace 'intercept' with 'Check error classes' in HiveQuerySuite.

### Why are the changes needed?
The changes improve the error framework.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
By running the modified test suite:
```
$ build/sbt "test:testOnly *HiveQuerySuite"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@panbingkun The test `lookup hive UDF in another thread` fails on my laptop locally:
```
Exception in thread "test" org.scalatest.exceptions.TestFailedException: Map("rawName" -> "not_a_udf", "fullName" -> "spark_catalog.a.not_a_udf") did not equal Map("rawName" -> "not_a_udf", "fullName" -> "spark_catalog.default.not_a_udf")
	at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:472)
	at org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:471)
	at org.scalatest.Assertions$.newAssertionFailedException(Assertions.scala:1231)
	at org.scalatest.Assertions$AssertionsHelper.macroAssert(Assertions.scala:1295)
	at org.apache.spark.SparkFunSuite.checkError(SparkFunSuite.scala:317)
	at org.apache.spark.SparkFunSuite.checkError(SparkFunSuite.scala:355)
	at org.apache.spark.sql.hive.execution.HiveQuerySuite$$anon$1.run(HiveQuerySuite.scala:1451)
[info] - lookup hive UDF in another thread *** FAILED *** (156 milliseconds)
[info]   success was false (HiveQuerySuite.scala:1457)
```

**Comment by panbingkun** [comment]
> @panbingkun The test `lookup hive UDF in another thread` fails on my laptop locally:
> 
> ```
> Exception in thread "test" org.scalatest.exceptions.TestFailedException: Map("rawName" -> "not_a_udf", "fullName" -> "spark_catalog.a.not_a_udf") did not equal Map("rawName" -> "not_a_udf", "fullName" -> "spark_catalog.default.not_a_udf")
> 	at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:472)
> 	at org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:471)
> 	at org.scalatest.Assertions$.newAssertionFailedException(Assertions.scala:1231)
> 	at org.scalatest.Assertions$AssertionsHelper.macroAssert(Assertions.scala:1295)
> 	at org.apache.spark.SparkFunSuite.checkError(SparkFunSuite.scala:317)
> 	at org.apache.spark.SparkFunSuite.checkError(SparkFunSuite.scala:355)
> 	at org.apache.spark.sql.hive.execution.HiveQuerySuite$$anon$1.run(HiveQuerySuite.scala:1451)
> [info] - lookup hive UDF in another thread *** FAILED *** (156 milliseconds)
> [info]   success was false (HiveQuerySuite.scala:1457)
> ```

I have fixed it.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala [hunk 1]
```diff
@@ -73,9 +72,17 @@ class HiveQuerySuite extends HiveComparisonTest with SQLTestUtils with BeforeAnd
     }
   }
 
-  private def assertUnsupportedFeature(body: => Unit): Unit = {
-    val e = intercept[ParseException] { body }
-    assert(e.getMessage.toLowerCase(Locale.ROOT).contains("operation not allowed"))
+  private def assertUnsupportedFeature(
+    body: => Unit,
+    message: String,
+    expectedContext: ExpectedContext): Unit = {
```
Comment (MaxGekk) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala:78]: 4 spaces, see https://github.com/databricks/scala-style-guide#spacing-and-indentation:
```suggestion
      body: => Unit,
      message: String,
      expectedContext: ExpectedContext): Unit = {
```

**Review by MaxGekk — APPROVED** [review]
LGTM except of a comment.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**